Spigot-Server-Patches/0697-Add-ElderGuardianAppearanceEvent.patch

```diff
@@ -0,0 +1,57 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Owen1212055 <23108066+Owen1212055@users.noreply.github.com>
+Date: Fri, 19 Mar 2021 23:39:09 -0400
+Subject: [PATCH] Add ElderGuardianAppearanceEvent
+
+
+diff --git a/src/main/java/net/minecraft/world/entity/monster/EntityGuardianElder.java b/src/main/java/net/minecraft/world/entity/monster/EntityGuardianElder.java
+index 9bd6f5d988c1e50e1bb729c407178b9cbcc62115..326cabe208431687be73f7da923037b87f1e6bac 100644
+--- a/src/main/java/net/minecraft/world/entity/monster/EntityGuardianElder.java
++++ b/src/main/java/net/minecraft/world/entity/monster/EntityGuardianElder.java
+@@ -72,6 +72,8 @@ public class EntityGuardianElder extends EntityGuardian {
+             boolean flag1 = true;
+             boolean flag2 = true;
+             boolean flag3 = true;
++            // Paper start
++            if (org.bukkit.craftbukkit.event.CraftEventFactory.handleGuardianAppearanceEvent(this, list)) {
+             Iterator iterator = list.iterator();
+ 
+             while (iterator.hasNext()) {
+@@ -82,6 +84,8 @@ public class EntityGuardianElder extends EntityGuardian {
+                     entityplayer.addEffect(new MobEffect(mobeffectlist, 6000, 2), org.bukkit.event.entity.EntityPotionEffectEvent.Cause.ATTACK); // CraftBukkit
+                 }
+             }
++            }
++            // Paper end
+         }
+ 
+         if (!this.ez()) {
+diff --git a/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java b/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java
+index 926440e846eff2c1aaa262aa2b3975b7dd225332..b47ef0dc21a1955a6d282e36d727bb0c9d619004 100644
+--- a/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java
++++ b/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java
+@@ -13,6 +13,8 @@ import java.util.List;
+ import java.util.Map;
+ import java.util.stream.Collectors;
+ import javax.annotation.Nullable;
++
++import io.papermc.paper.event.entity.ElderGuardianAppearanceEvent;
```

Comment:
This import may have to be marked with the `// Paper` suffix or removed in favour of a fully qualified reference to the event down the line.  All but the `io.papermc.paper.event.block.BlockFailedDispenseEvent` import follow that precedent.